nilearn/masking.py

```diff
@@ -705,8 +720,8 @@ def compute_multi_gray_matter_mask(target_imgs, threshold=.5,
         pass
 
     mask = compute_brain_mask(target_imgs[0], threshold=threshold,
-                                    connected=connected, opening=opening,
-                                    memory=memory, verbose=verbose)
+                              connected=connected, opening=opening,
+                              memory=memory, verbose=verbose, mask_type='gm')
```

Comment:
@alpinho In `nilearn/input_data/tests/test_multi_nifti_masker.py::test_compute_multi_gray_matter_mask`, this line gets executed, which triggers the call to `fetch_icbm152_brain_gm_mask()`

Reply:
Yes, I added that intentionally since in this function we want to use the estimated the grey-matter mask. Isn't that correct?

Reply:
I think this is correct, I am just saying that this is how some pre-existing tests are calling the fetch function you added in `compute_brain_mask`. 
I am not sure, but I think the solution would be to write a function similar to `load_mni152_brain_mask()` but for grey and white matter masks. That is, only based on the template that we ship with nilearn. Does this make sense?

Reply:
@alpinho it looks like this test (`test_compute_multi_gray_matter_mask`) is failing again. 

It computes a gm mask on some random data with a template strategy and compare the output to some ground truth:

https://github.com/nilearn/nilearn/blob/9a05216346e127212b765bd0ad4797837c3f1898/nilearn/input_data/tests/test_multi_nifti_masker.py#L167-L184

It was introduced a while ago in #1682. I'm not sure where the expected output comes from but I think it assumes a binary erosin with 2 iterations on an array full of ones, which comes from the call to `_post_process_mask` after the resampling and thresholding:

https://github.com/nilearn/nilearn/blob/9a05216346e127212b765bd0ad4797837c3f1898/nilearn/masking.py#L171-L176

Since we are now loading a different template (gray matter instead of systematically whole brain), the input to `_post_process_mask` is different and the binary erosion is zeroing the thresholded mask.

What I don't understand is that the function being tested is supposed to compute the gray matter mask and was using, prior to this PR, the whole brain template (there is a comment from @KamalakerDadi in #1682 saying the same thing...).

I'm not really sure what we should do here. One way to keep the test as is would be to change the default threshold value of `compute_multi_gray_matter_mask` from 0.5 to 0, but I'm not sure we want to do that. Another solution would be to adapt the test to the new (and correct) behavior.

I hope this is understandable. Let me know what you think (also pinging @bthirion here).

Reply:
Thanks @NicolasGensollen for your thoughts. I was yesterday looking to the same lines of `nilearn/nilearn/input_data/tests/test_multi_nifti_masker.py` and could not understand well the underlying reason of these tests. Just one quick clarification about the motivation of this PR. It comes from a previous PR, #2459, where we renamed the function `nilearn.masking.compute_gray_matter_mask` to `nilearn.masking.compute_brain_mask` because in the issue #2280 someone reported that `nilearn.masking.compute_gray_matter_mask` was outputting the whole-brain mask instead of the gray-matter mask. Given this, I can understand why the test evaluates gray-matter mask's estimation. Thus, I also prefer to ask @bthirion how we should change the test functions.

Reply:
We should avoid tweaking default parameters of too many functions...
Does Changing the erosion parameter to 1 in the text solve the issue ?

Reply:
this seems to have been resolved?